### ISSUE TITLE
dev branch / branch for new major release

### ISSUE THREAD

**Issue by derNarr** [quantling/pyndl#104]
I would suggest to use a branch for the next major release. For now `0.4.x`. All pull request that change the API or are not backwards compatible will be merged into that branch. All other pull request will be merged directly into `master`.
This way the difference between `master` and the next major release should be kept as small as possible.
Any other ideas? Arguments?

**Comment by dekuenstle** [comment]
TL;DR; Great idea. Other people use similar successfully. We can talk about minor changes towards git-flow.

I totaly agree, that we have to introduce more branches! 
Some people call a work flow similar to what we do atm [github-flow](http://scottchacon.com/2011/08/31/github-flow.html). 
They say it's appropriate if you don't do "releases" and have a stable master all the time as a leaner *git-flow* workflow. 
Especially because we want to do releases, we should look towards git-flow like you did.
Your suggestion can be extended by a develop-branch and hotfix-branches to do full [git-flow](http://nvie.com/posts/a-successful-git-branching-model/).
